src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+import randoop.sequence.Sequence;
+import randoop.sequence.SequenceCollection;
+import randoop.test.DummyCheckGenerator;
+import randoop.types.NonParameterizedType;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.EquivalenceChecker;
+import randoop.util.Randomness;
+import randoop.util.SimpleArrayList;
+import randoop.util.SimpleList;
+
+/**
+ * A demand-driven approach to construct inputs. Randoop works by selecting a method, then trying to
+ * find inputs to that method. Ordinarily, Randoop works bottom-up: if Randoop cannot find inputs
+ * for the selected method, it gives up and selects a different method. This demand-driven approach
+ * works top-down: if Randoop cannot find inputs for the selected method, then it looks for methods
+ * that create values of the necessary type, and recursively tries to call them.
+ *
+ * <p>The demand-driven approach implements the "Detective" component described by the paper "GRT:
+ * Program-Analysis-Guided Random Testing" by Ma et. al (appears in ASE 2015):
+ * https://people.kth.se/~artho/papers/lei-ase2015.pdf .
+ */
+public class DemandDrivenInputCreation {
+
+  /**
+   * The set of classes that are specified by the user for Randoop to consider. These are classes
+   * supplied in the command line arguments (e.g. --classlist).
+   */
+  private static Set<@ClassGetName String> SPECIFIED_CLASSES =
+      GenInputsAbstract.getClassnamesFromArgs(AccessibilityPredicate.IS_ANY);
+
+  /**
+   * The set of classes that demand-driven uses to generate inputs but are not specified by the
+   * user.
+   */
+  private static Set<Class<?>> unspecifiedClasses = new LinkedHashSet<>();
+
+  /** True if an exact type match is required. */
+  private static boolean EXACT_TYPE_MATCH;
+
+  /** If true, only return sequences that are appropriate to use as a method call receiver. */
+  private static boolean ONLY_RECEIVERS;
+
+  // TODO: The original paper uses a "secondary object pool (SequenceCollection in Randoop)"
+  // to store the results of the demand-driven input creation. This theorectically reduces
+  // the search space for the missing types. Consider implementing this feature and test whether
+  // it improves the performance.
+
+  /**
+   * Performs a demand-driven approach for constructing input objects of a specified type, when the
+   * sequence collection contains no objects of that type.
+   *
+   * <p>This method internally identifies a set of methods/constructors that return objects of the
+   * required type. For each of these methods: it generates a method sequence for the method by
+   * recursively searching for necessary inputs from the provided sequence collection; executes it;
+   * and if successful, stores the sequence in the sequence collection for future use.
+   *
+   * <p>Finally, it returns the newly-created sequences.
+   *
+   * <p>Invariant: This method is only called where the component manager lacks an object that is of
```

Comment:
Mention the relationship between the component mananger and the `sequenceCollection` formal parameter.